raspberry/checkout/checkout_main.py

```diff
@@ -62,16 +65,40 @@ def process_rfid_card(self, uid):
             print("Item already scanned")
             self.interactions.buzz()
             self.interactions.indicate_error()
+
+    def cancel_action(self):
+        if self.state == 1:
+            return self.cancel_checkout
+        else:
+            return self.logic.remove_current_tag
+
+    def confirm_action(self):
+        if self.state == 1:
+            return self.finish_checkout
+        else:
+            def change_state():
+                self.state = 1
+            return change_state
+
+    def next(self):
+        self.logic.next_product()
+        self.interactions.display_product_details(self.logic.get_current_product())
+
+    def prev(self):
+        self.logic.previous_product()
+        self.interactions.display_product_details(self.logic.get_current_product())
```

Comment:
```suggestion
    def prev(self):
        if self.state == 1:
            self.logic.previous_product()
            self.interactions.display_product_details(self.logic.get_current_product())
```
we shouldn't really be able to do anything with encoder while scanning the items